src/particleRepresentation/ParticleContainer.h

```diff
@@ -0,0 +1,98 @@
+//
+// Created by daniel on 24.04.24.
+//
+
+#pragma once
+
+#include "particleRepresentation/particle/Particle.h"
+
+#include <vector>
+#include <iterator>
+
+/**
+ * @brief Class to store the particles for simulation.
+ *
+ * The storage of the particles is based on std::vector. This guarantees fast iteration over the particles, because
+ * they are stored consecutive in memory.
+ */
+
+class ParticleContainer {
+private:
+    std::vector<Particle> particles;
+
+public:
+
+    ParticleContainer() = default;
+
+    /**
+     * @brief Create a ParticleContainer which can store up to capacity particles.
+     *
+     * @param capacity
+     *
+     * If the capacity is exceeded the whole data structure will be extended leading to reallocation.
+     */
+    explicit ParticleContainer(size_t capacity);
```

Comment:
Even though it is not used yet, I like that you are thinking about memory management.